component/cmp_difficulty.go

```diff
@@ -0,0 +1,163 @@
+package component
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+)
+
+type Difficulty struct {
+	parent    *Subcategory
+	ID        string `json:"-"`
+	Name      string `json:"name"`
```

Comment:
Name should named description and we need a property named "key" or "name" which would have one of the three options, beginner, advanced, expert.